api/src/services/healthService.js

```diff
@@ -1,140 +1,261 @@
+      }
     }
+  });
+
+  res.json(
+    healthErrors.length
+      ? { msg: 'Health check failed', errors: healthErrors }
+      : { msg: `Validated ${validatedAccountsNumber} account(s)` }
+  );
+}
+
+async function liveness(req, res) {
+  res.status(200).json({
+    msg: 'Liveness probe is ok',
+  });
+}
+
+async function readiness(req, res) {
+  let healthErrors = [];
+  let client;
+  try {
+    client = await getClient();
   } catch (error) {
-    res.status(400).json({
-      msg: error.message,
+    console.error(error);
+    healthErrors.push(`Unable to connect to database:${err.message}`);
+  } finally {
+    client?.end();
+  }
+
+  if (!cacheService.initialized()) {
+    healthErrors.push('Cache service is not initialized yet');
+  }
+
+  if (!cereNetworkService.initialized()) {
+    healthErrors.push('Cere network service is not initialized yet');
+  }
+
+  if (!ethNetworkService.initialized()) {
+    healthErrors.push('Ethereum network service is not initialized yet');
+  }
+  
+  healthErrors.length
+    ? res.status(503).json({ msg: 'Readiness probe failed', errors: healthErrors })
+    : res.status(200).json({ msg: 'Readiness probe is ok' });
+}
+
+async function checkBlockchainHealth(req, res) {
+  const { query } = req;
+  let networks = splitParams(query.networks);
+  
+  const validationErrors = validateNetworks(networks);
+  if (validationErrors.length) {
+    return res.status(400).json({
+      msg: 'Validation failed',
+      errors: validationErrors
     });
   }
+
+  let healthErrors = [];
+  await Promise.all(networks.map(async network => {
+    try {
+      await cereNetworkService.getHealth(network);
+    } catch (error) {
+      console.error(error);
+      healthErrors.push(`Unable to connect to ${network} network`);
+    }
+  }));
+
+  res.status(200).json(
+    healthErrors.length
+      ? { msg: 'Health check failed', errors: healthErrors }
+      : { msg: `${networks.join(',')} status is ok`}
+  );
+}
+
+async function checkBlockchainBlocksFinalization(req, res) {
+  const { query } = req;
+  let networks = splitParams(query.networks);
+  const cereConfig = blockchains.find(blockchain => blockchain.name === blockchainNames.CERE);
+  
+  const validationErrors = validateNetworks(networks);
+  if (validationErrors.length) {
+    return res.status(400).json({
+      msg: 'Validation failed',
+      errors: validationErrors
+    });
+  }
+  
+  let healthErrors = [];
+  await Promise.all(
+    networks.map(async network => {      
+      const [best, finalized] = await Promise.all([
+        cereNetworkService.getFinalizedBlock(network),
+        cereNetworkService.getBestBlock(network)
+      ]);      
+      const bestBlockNumberDiff = best - finalized;
+      if (bestBlockNumberDiff > cereConfig.bestBlockNumberDiff) {
```

Comment:
`cereConfig.bestBlockNumberDiff` should be updated in the k8s configs?

Reply:
yes